### PR TITLE
Breaks if no package.json is present

### PR DESCRIPTION
**Pull request by JimmyBoh**
If there is no `package.json` available then the path is `null` and you can't stop it from breaking.

Even if you manually provide the `projectName` in options it will still break. This adds a few simple checks to prevent it from blowing up when you do provide the name.

I've also added an appropriate test, which properly fails when run without the changes to `index.js` and correctly passes once added.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -15,9 +15,13 @@ const parentDir = path.dirname(module.parent.filename);
...
 
-		if (!opts.projectName) {
+		// If the package.json was not found, avoid breaking with `require(null)`.
+		if (pkgPath) {
+			opts = Object.assign({projectName: require(pkgPath).name}, opts);
+		}
+
+		if (!opts || !opts.projectName) {
 			throw new Error('Project name could not be inferred. Please specify the `projectName` option.');
 		}
```
Comment (sindresorhus) [inline index.js]: I think I would rather do:

```js
const pkgPath = pkgUp.sync(parentDir);

opts = Object.assign({
	// if the package.json was not found, avoid breaking with `require(null)`.
	projectName: pkgPath && require(pkgPath).name
}, opts);

if (!opts.projectName) {
	throw new Error('Project name could not be inferred. Please specify the `projectName` option.');
}
```

**Comment by sindresorhus** [comment]
Looks good.

Just curious, why is there no package.json available?

File: index.js (hunk 1 above)
Comment (JimmyBoh) [inline index.js]: Tons of ways to handle this null value. I will update it to match yours, nice and clean.

Comment (sindresorhus) [inline index.js]: Yup, totally subjective. Both work equally fine :)

**Comment by sindresorhus** [comment]
Can you fix the merge conflict? Sorry, I had a local fix that I pushed by accident now.

**Comment by JimmyBoh** [comment]
This is limited to select installs of Electron apps (depending on how they are packaged) but I use [this one](https://github.com/szwacz/electron-boilerplate) by @szwacz. I'm not sure if there are any other downsides, but I figured an unavoidable exception like this could be a problem for others as well.

**Comment by JimmyBoh** [comment]
And no worries about the merge conflict. I'll fix that once I get home (done at the office for today).

**Comment by JimmyBoh** [comment]
I just noticed the test you added [still has `.only` on it](https://github.com/sindresorhus/conf/commit/622f648fed33166e331e8859c8320e395c6e5fad#diff-1dd241c4cd3fd1dd89c570cee98b79ddR150). Do you want to fix that before I merge? Or you want me to take care of that right now?

**Comment by sindresorhus** [comment]
[Ugh, not again.](https://github.com/sindresorhus/conf/issues/2#issuecomment-227813534) Fixed in master. Thanks for noticing.

**Comment by JimmyBoh** [comment]
No problem, happens to the best of us!

**Comment by sindresorhus** [comment]
Thank you :)

**Comment by JimmyBoh** [comment]
My pleasure!

Quick (potentially silly) question: This does not say "merged", but instead closed with a commit that "I committed with you". Is that the result of Github squashing the commits? Or something you did to keep things clean? Next time should I have squashed my commits first?

**Comment by SamVerschueren** [comment]
GitHub now has a button to squash and merge a PR so no need in doing that yourself, although you can off course :).

Not sure if Sindre uses the button though. Before there was a button to squash, he did everything "manually".

**Comment by JimmyBoh** [comment]
Hey @SamVerschueren, I've seen that but have yet to use it. I'm guessing he did things manually, hence the "closed" status as opposed to Github's "merged" status. Thanks for the insight!

**Comment by JimmyBoh** [comment]
Oh, and @sindresorhus, may I recommend a bump to [`electron-config`](https://github.com/sindresorhus/electron-config), so that it can have the latest updates as well (`"conf": "^0.10.0"` restricts it to `0.10.*`).

Thank you!

**Comment by sindresorhus** [comment]
> I've seen that but have yet to use it. I'm guessing he did things manually, hence the "closed" status as opposed to Github's "merged" status. Thanks for the insight!

Yeah. I did it manually as I wanted to tweak something on merge. I could probably have used squash and then force pushed, but old habits die hard.

> may I recommend a bump to electron-config

Done :)

**Comment by JimmyBoh** [comment]
Thank You!

:]

**Pull request closed without merging**